File: ios-app/ViaductEcho/Models/Article.swift
import Foundation

struct Article: Identifiable, Codable {
    let id: Int
    let title: String
    let link: String
    let summary: String
    let source: String
    let sourceType: String
    let publishedDate: Date
    let createdAt: Date
    let aiImageUrl: String?
    
    enum CodingKeys: String, CodingKey {
        case id
        case title
        case link
        case summary
        case source
        case sourceType = "source_type"
        case publishedDate = "published_date"
        case createdAt = "created_at"
        case aiImageUrl = "ai_image_url"
    }
}

struct ArticleDetail: Identifiable, Codable {
    let id: Int
    let title: String
    let link: String
    let summary: String
    let source: String
    let sourceType: String
    let publishedDate: Date
    let createdAt: Date
    let updatedAt: Date?
    let processed: Bool
    let extractedContent: String?
    let aiSummary: String?
    let aiImageUrl: String?
    
    enum CodingKeys: String, CodingKey {
        case id
        case title
        case link
        case summary
        case source
        case sourceType = "source_type"
        case publishedDate = "published_date"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case processed
        case extractedContent = "extracted_content"
        case aiSummary = "ai_summary"
        case aiImageUrl = "ai_image_url"
    }
}

struct PaginatedArticles: Codable {
    let articles: [Article]
    let pagination: PaginationInfo
}

struct PaginationInfo: Codable {
    let page: Int
    let perPage: Int
    let totalItems: Int
    let totalPages: Int
    let hasNext: Bool
    let hasPrev: Bool
    
    enum CodingKeys: String, CodingKey {
        case page
        case perPage = "per_page"
        case totalItems = "total_items"
        case totalPages = "total_pages"
        case hasNext = "has_next"
        case hasPrev = "has_prev"
    }
}

File: android-app/app/src/main/java/com/viaductecho/android/data/models/Article.kt
package com.viaductecho.android.data.models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Article(
    val id: Int,
    val title: String,
    val link: String,
    val summary: String,
    val source: String,
    @SerializedName("source_type")
    val sourceType: String,
    @SerializedName("published_date")
    val publishedDate: String?,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("image_url")
    val imageUrl: String?,
    @SerializedName("ai_summary")
    val aiSummary: String?,
    @SerializedName("extracted_content")
    val extractedContent: String?,
    val processed: Boolean? = null,
    @SerializedName("updated_at")
    val updatedAt: String? = null
) : Parcelable

@Parcelize
data class ArticlesResponse(
    val articles: List<Article>,
    val pagination: Pagination
) : Parcelable

@Parcelize
data class Pagination(
    val page: Int,
    @SerializedName("per_page")
    val perPage: Int,
    @SerializedName("total_items")
    val totalItems: Int,
    @SerializedName("total_pages")
    val totalPages: Int,
    @SerializedName("has_next")
    val hasNext: Boolean,
    @SerializedName("has_prev")
    val hasPrev: Boolean
) : Parcelable

@Parcelize
data class Source(
    val name: String,
    @SerializedName("article_count")
    val articleCount: Int,
    @SerializedName("processed_count")
    val processedCount: Int,
    @SerializedName("latest_article")
    val latestArticle: String?
) : Parcelable

@Parcelize
data class SourcesResponse(
    val sources: List<Source>,
    @SerializedName("total_sources")
    val totalSources: Int
) : Parcelable

// Health check response
@Parcelize
data class HealthResponse(
    val status: String,
    @SerializedName("total_articles")
    val totalArticles: Int?,
    @SerializedName("recent_articles_24h")
    val recentArticles24h: Int?,
    @SerializedName("database_connected")
    val databaseConnected: Boolean,
    val timestamp: String,
    val error: String? = null
) : Parcelable
